subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -909,8 +910,9 @@ public TestFramework getTestFramework() {
     public TestFramework testFramework(@Nullable Closure testFrameworkConfigure) {
```

Comment:
So the `testFramework` property is not accessed until analyzing task inputs just prior to executing the `Test` task, correct?

Reply:
It depends on if the task has any configuration for the test framework, but in general:

1. We look at the test framework property once when building the task graph. This is because it's marked `@Nested` and we look at the value for other inputs (test framework options)
2. We look at the test framework if the user calls `options {}`
3. We look at the test framework just before the task starts executing. The property is also finalized at this point.